src/core/figures.js

```diff
@@ -12,33 +14,30 @@ export function run(conf) {
   // process all figures
   const figMap = {};
   const tof = [];
-  let num = 0;
-  $("figure").each(function() {
-    const $fig = $(this);
-    const $cap = $fig.find("figcaption");
-    const tit = $cap.text();
-    const id = $fig.makeID("fig", tit);
-    if (!$cap.length)
-      pub("warn", "A `<figure>` should contain a `<figcaption>`.");
+  document.querySelectorAll("figure").forEach((fig, i) => {
+    const caption = fig.querySelector("figcaption");
+    const title = caption.textContent;
+    const id = addId(fig, "fig", title);
 
-    // set proper caption title
-    num++;
-    $cap
-      .wrapInner($("<span class='fig-title'/>"))
-      .prepend(document.createTextNode(" "))
-      .prepend($("<span class='figno'>" + num + "</span>"))
-      .prepend(document.createTextNode(conf.l10n.fig));
-    figMap[id] = $cap.contents();
-    const $tofCap = $cap.clone();
-    $tofCap
-      .find("a")
-      .renameElement("span")
-      .removeAttr("href");
+    if (caption) {
+      // set proper caption title
+      wrapInner(caption, hyperHTML`<span class='fig-title'>`);
+      caption.prepend(
+        conf.l10n.fig,
+        hyperHTML`<span class='figno'>${i + 1}</span>`,
+        " "
+      );
+    } else pub("warn", "A `<figure>` should contain a `<figcaption>`.");
```

Comment:
If we are emitting a warning here, maybe also add the relevant `<figure>` in `console.warn` to make it easier to identify

Reply:
Maybe we can directly link in the warning list UI as link-to-dfn does:

https://github.com/w3c/respec/blob/c1a1f82e2bebd8e75bbcc809e8afebc7c1fee13c/src/core/link-to-dfn.js#L62-L67

I hope we have a general helper function for this.

Reply:
The best would be pointing the line from the source, which is impossible 😂

Reply:
Filed #1944

Reply:
For now, maybe just do: 
```JS
else {
   const msg = "A `<figure>` should contain a `<figcaption>`".
   pub("warn", `${msg}. See developer console for offending element.`);
   elem.classList.add("respec-offending-element");
   console.warn(msg, elem);
}

```